src/psim/core/state.cpp

```diff
@@ -40,18 +40,64 @@ bool State::has_writable(std::string const &name) const {
   return _writable_fields.count(name);
 }
 
-void State::add_writable(std::string const &name, StateFieldWritableBase *field) {
-  if (has(name))
-    throw std::runtime_error("Registered two or more state fields under the name: " + name);
+void State::add_writable(StateFieldWritableBase *field) {
+  {
+    // Check if we have a name collision with existing readable fields
+    auto const iter = _readable_fields.find(field->name());
+    if (iter != _readable_fields.end()) {
+      auto const *_field = iter->second;
+      throw std::runtime_error(
+          "'" + _field->name() + ":" + _field->type() + "' already registered " +
+          "as a readable field. Cannot add '" + field->name() + ":" +
+          field->type() + "' as a writable field"
+        );
```

Comment:
Here's an example of what some of the new error messages look like just to give an idea.